projects/core/components/primitive-textfield/textfield/textfield.component.ts

```diff
@@ -0,0 +1,69 @@
+import {
+    ChangeDetectionStrategy,
+    Component,
+    DoCheck,
+    ElementRef,
+    HostBinding,
+    Inject,
+} from '@angular/core';
+import {TuiIdService} from '@taiga-ui/cdk';
+import {
+    TEXTFIELD_CONTROLLER_PROVIDER,
+    TUI_DESCRIBED_BY_PROVIDERS,
+    TUI_TEXTFIELD_WATCHED_CONTROLLER,
+    TuiDescribedByDirective,
+    TuiTextfieldController,
+} from '@taiga-ui/core/directives';
+import {TuiTextfieldHost} from '@taiga-ui/core/interfaces';
+import {TUI_TEXTFIELD_HOST} from '@taiga-ui/core/tokens';
+
+// @dynamic
+@Component({
+    selector: 'input[tuiTextfield]',
+    template: '',
+    providers: [
+        TuiDescribedByDirective,
+        TUI_DESCRIBED_BY_PROVIDERS,
+        TEXTFIELD_CONTROLLER_PROVIDER,
+    ],
+    host: {
+        type: 'text',
+        '[attr.aria-placeholder]': 'controller.exampleText',
+        '[attr.aria-invalid]': 'host.invalid',
+        '[attr.disabled]': 'host.disabled || null',
+        '[tabIndex]': 'host.focusable ? 1 : 0',
+        '[readOnly]': 'host.readOnly',
+        '[value]': 'host.value',
+        '(input)': 'host.onValueChange($event.target.value)',
+    },
+    styleUrls: ['textfield.style.less'],
+    changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class TuiTextfieldComponent implements DoCheck {
+    constructor(
+        @Inject(TUI_TEXTFIELD_HOST) readonly host: TuiTextfieldHost,
+        @Inject(TUI_TEXTFIELD_WATCHED_CONTROLLER)
+        readonly controller: TuiTextfieldController,
+        @Inject(ElementRef) private readonly elementRef: ElementRef<HTMLInputElement>,
+        @Inject(TuiDescribedByDirective)
+        private readonly describedBy: TuiDescribedByDirective,
+        @Inject(TuiIdService)
+        private readonly idService: TuiIdService,
+    ) {
+        this.host.process(this.elementRef.nativeElement);
+    }
+
+    @HostBinding('attr.aria-describedby')
+    get computedDescribedBy(): string | null {
+        return this.describedBy.computedDescribedBy;
+    }
+
+    @HostBinding('id')
```

Comment:
do we really need to change the id?

```ts
 @Component({
        template: `
            <input id="my-cystom-id-for-cypress" tuiTextfield />
        `,
    })
```

what will happen with my id?

Reply:
I've added a unit test specifically for this :) It works with all 3 cases - auto id when there is no id, keeps your case with static id and allows binding custom id.